### PR TITLE
Added Tests for testing the content rendered on the page library

### LINKED ISSUE #425: Increased test coverage on Library page
## ✨ Feature request
A test for checking out content rendered on the Library page, so you can tell if something is rendered incorrectly

### 📜 Acceptance Criteria

- [ ] A test that passes for all commponents 

- [ ] The test should check out the content inside each component

### PR DESCRIPTION
**Pull request by te4-andre-skvarc**
## ⚙️ What issue did you implement or fix?
Fixes #425 

## 📜 Summary
- Added a simple test to check the rendered content on the library page.

## 📝 Type of change

- [x] ✨ New feature or enhancement (non-breaking change which adds functionality)

## ✅ Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Review by itggot-linus-styren — CHANGES_REQUESTED** [review]
Nice TDD, I like what I see. Please don't commit all the "examples", makes this PR hard to review. Also looks like you ran cypress from the incorrect folder.

Reminder to run git pull --rebase origin develop before creating PR to avoid complaints about problems already solved. This time you can just ignore it, since it will not be a problem on develop branch.

**Review by itggot-linus-styren — APPROVED** [review]
LGTM, maybe some more functionality could be tested in the future. Consider reading this article: https://blog.logrocket.com/how-to-write-useful-end-to-end-tests-with-cypress/

You may need to change the title from DataTable to table to not break other tests, this was changed here by @te4-adrian-almetun-smeds :
https://github.com/itggot-TE4/Yabs/blame/develop/frontend/src/components/ListComponent.vue#L2

File: frontend/src/components/ListComponent.vue [hunk 1]
```diff
@@ -1,5 +1,5 @@
 <template>
-  <v-card>
+  <v-card data-cy="DataTable">
```
Comment (te4-adrian-almetun-smeds) [inline frontend/src/components/ListComponent.vue:2]: Rebase on the dev branch in order to get the table data tag

File: frontend/cypress/integration/LibraryTest.js [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+    it('Checks scan the book barcode card' , () => {
+        cy.get('[data-cy=SecondCard]')
+        cy.contains('Scan the book barcode')
+        cy.contains('Bokens Streckkod')
+        cy.contains('Låna Ut')    
+    })
+    
+    it('Checks the data tables' , () => {
+        cy.get('[data-cy=DataTable]')
```
Comment (te4-adrian-almetun-smeds) [inline frontend/cypress/integration/LibraryTest.js:24]:
```suggestion
        cy.get('[data-cy=table]')
```

Here we can use the `data-cy` tag that has already been defined.

File: frontend/src/components/ReturnLoanComponent.vue [hunk 3]
```diff
@@ -1,6 +1,6 @@
 <template>
   <div>
-    <v-card>
+    <v-card data-cy="FirstCard">
```
Comment (te4-adrian-almetun-smeds) [inline frontend/src/components/ReturnLoanComponent.vue:3]: As this is a reusable component, maybe we should name the data tag something related to the component name. Perhaps `returnLoanCard` can be used.

File: frontend/src/components/LibraryLoanComponent.vue [hunk 4]
```diff
@@ -1,6 +1,6 @@
 <template>
   <div>
-    <v-card>
+    <v-card data-cy="SecondCard">
```
Comment (te4-adrian-almetun-smeds) [inline frontend/src/components/LibraryLoanComponent.vue:3]: As this is a reusable component, maybe we should name the data tag something related to the component name. Perhaps `libraryLoanCard` can be used.

**Review by te4-adrian-almetun-smeds — CHANGES_REQUESTED** [review]
This commit will indeed break our other test due to it overwriting the data tag on the library card.

See my comments to resolve this.

File: frontend/src/components/ListComponent.vue (hunk 1 above)
Comment (Linkaan) [reply frontend/src/components/ListComponent.vue:2]: I suggest to simply cherry-pick 4b3e159a0a2006e7e777db1d47dc7fe0becdb51a instead and then resolve conflict.

File: frontend/src/components/LibraryLoanComponent.vue (hunk 4 above)
Comment (te4-david-jensen) [inline frontend/src/components/LibraryLoanComponent.vue:3]:
```suggestion
    <v-card data-cy="borrowCard">
```

File: frontend/cypress/integration/LibraryTest.js [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+    it('Checks scan your book barcode card' , () => {
+        cy.get('[data-cy=FirstCard]')
+        cy.contains('Scan your book barcode')
+        cy.contains('Bokens Streckkod')
+        cy.contains('Return')  
+        
+    })
+
+    it('Checks scan the book barcode card' , () => {
+        cy.get('[data-cy=SecondCard]')
```
Comment (te4-david-jensen) [inline frontend/cypress/integration/LibraryTest.js:17]:
```suggestion
        cy.get('[data-cy=borrowCard]')
```

File: frontend/cypress/integration/LibraryTest.js [hunk 6]
```diff
@@ -0,0 +1,40 @@
+import baseVisit from '../helper';
+
+describe('This test checks the rendered content' , () => {
+    it('checks the whole page', () => {
+      baseVisit('/library');
+    });
+
+    it('Checks scan your book barcode card' , () => {
+        cy.get('[data-cy=FirstCard]')
```
Comment (te4-david-jensen) [inline frontend/cypress/integration/LibraryTest.js:9]:
```suggestion
        cy.get('[data-cy=returnCard]')
```

File: frontend/src/components/ReturnLoanComponent.vue (hunk 3 above)
Comment (te4-david-jensen) [inline frontend/src/components/ReturnLoanComponent.vue:3]:
```suggestion
    <v-card data-cy="returnCard">
```

File: frontend/src/components/LibraryLoanComponent.vue (hunk 4 above)
Comment (te4-david-jensen) [reply frontend/src/components/LibraryLoanComponent.vue:3]: 👍

File: frontend/src/components/ReturnLoanComponent.vue (hunk 3 above)
Comment (te4-david-jensen) [reply frontend/src/components/ReturnLoanComponent.vue:3]: 👍

**Review by te4-david-jensen — APPROVED** [review]
Looks good! 👍  :shipit:

File: frontend/cypress/integration/createTitleAndBooks.js [hunk 7]
```diff
@@ -13,7 +13,7 @@ describe('This test will create a title and a book', () => {
     const bookTitle = 'Book made from Test'
     const isbn = '420-420'
     const cost = '2912'
-
+
```
Comment (te4-adrian-almetun-smeds) [inline frontend/cypress/integration/createTitleAndBooks.js:16]:
```suggestion

```

File: frontend/cypress/integration/LibraryTest.js [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+        cy.contains('Titel')
+        cy.contains('Kostnad')
+        cy.contains('Typ')
+        cy.contains('ISBN')
+        
+    })
+    
+    
+
+
```
Comment (te4-adrian-almetun-smeds) [inline frontend/cypress/integration/LibraryTest.js:39]:
```suggestion
```

File: frontend/cypress/integration/LibraryTest.js [hunk 9]
```diff
@@ -0,0 +1,40 @@
...
+    it('checks the whole page', () => {
+      baseVisit('/library');
+    });
+
+    it('Checks scan your book barcode card' , () => {
+        cy.get('[data-cy=returnCard]')
+        cy.contains('Scan your book barcode')
+        cy.contains('Bokens Streckkod')
+        cy.contains('Return')  
+
```
Comment (te4-adrian-almetun-smeds) [inline frontend/cypress/integration/LibraryTest.js:13]:
```suggestion
```

File: frontend/cypress/integration/LibraryTest.js [hunk 10]
```diff
@@ -0,0 +1,40 @@
...
+        cy.contains('Search')
+        cy.contains('Lånad av')
+        cy.contains('Utlånad av')
+        cy.contains('Boktitel')
+        cy.contains('Utgångsdatum')
+        cy.contains('Titel')
+        cy.contains('Kostnad')
+        cy.contains('Typ')
+        cy.contains('ISBN')
+
```
Comment (te4-adrian-almetun-smeds) [inline frontend/cypress/integration/LibraryTest.js:34]:
```suggestion
```

**Review by te4-adrian-almetun-smeds — CHANGES_REQUESTED** [review]
Looks good. Just remove some empty lines and we are good to merge! :+1:

**Review by te4-adrian-almetun-smeds — APPROVED** [review]
LGTM 👍

**Pull request merged**